### PR TITLE
feat: Robust handling of ROS GPG key

### PR DESCRIPTION
**Pull request by ar13pit**
- This should be backwards compatible with existing ros sources file

### CODE REVIEW

**Comment by ar13pit** [comment]
This of course needs the addition of the new tue logging commands

File: ros-setup/install.bash [hunk 1]
```diff
@@ -1,21 +1,63 @@
...
+FORCE_UPDATE=false
 
-    tue-install-apt-get-update
-    tue-install-debug "Added ROS sources to apt-get successfully"
+if [[ ! -f "${sourcefile}" ]]
+then
+    tue-install-debug "Adding ROS sources to apt-get"
+    ADD_ROS_SOURCES=true
+    ADD_ROS_GPG_KEY=true
+elif [[ $(cat "${sourcefile}") != "${sourceurl}" ]]
```
Comment (MatthijsBurgh) [inline ros-setup/install.bash:19]: Better to do a grep here as there could be any comments or so in the file.

File: ros-setup/install.bash [hunk 2]
```diff
@@ -1,21 +1,63 @@
...
+    tue-install-debug "Successfully added/updated ROS source file in apt"
+
+    FORCE_UPDATE=true
+fi
+
+if [[ "${FORCE_UPDATE}" == "true" ]]
 then
-    tue-install-echo "Updating expired GPG key"
-    tue-install-pipe sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+    sudo apt-get update -qq
```
Comment (MatthijsBurgh) [inline ros-setup/install.bash:61]: Use `tue-install-apt-get-update`

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Might include a non debug print statement

**Comment by MatthijsBurgh** [comment]
ping @ar13pit

File: ros-setup/install.bash (hunk 1 above)
Comment (ar13pit) [reply ros-setup/install.bash:19]: The idea is if the files are not identical then overwrite it

**Review by MatthijsBurgh — APPROVED** [review]

**Pull request merged**